legacy-sdk/whirlpool/tests/utils/litesvm.ts

```diff
@@ -0,0 +1,1230 @@
+/**
+ * LiteSVM utilities for migrating tests from solana-test-validator
+ */
+import type { AnchorProvider } from "@coral-xyz/anchor";
+import * as anchor from "@coral-xyz/anchor";
+import {
+  PublicKey,
+  Keypair,
+  Transaction,
+  VersionedTransaction,
+} from "@solana/web3.js";
+import { LiteSVM, Clock } from "litesvm";
+import bs58 from "bs58";
+import * as fs from "fs";
+import * as path from "path";
+import {
+  NATIVE_MINT,
+  NATIVE_MINT_2022,
+  TOKEN_PROGRAM_ID,
+  TOKEN_2022_PROGRAM_ID,
+} from "@solana/spl-token";
+import { TEST_TOKEN_PROGRAM_ID } from "./test-consts";
+
+let _litesvm: LiteSVM | null = null;
+
+// Event listener management for onLogs simulation
+interface LogListener {
+  id: number;
+  filter: any;
+  callback: (logs: any, ctx: any) => void;
+}
+
+let _logListeners: LogListener[] = [];
+let _nextListenerId = 1;
+
+// Transaction history storage for getParsedTransaction
+interface TransactionRecord {
+  signature: string;
+  logs: string[];
+  slot: number;
+  blockTime: number;
+}
+let _transactionHistory: Map<string, TransactionRecord> = new Map();
+
+function ensureNativeMintAccounts(litesvm: LiteSVM) {
+  // Minimal mint layout: set owner and 82-byte data buffer
+  const ensure = (mint: PublicKey, owner: PublicKey) => {
+    const acc = litesvm.getAccount(mint);
+    if (!acc) {
+      const data = Buffer.alloc(82, 0);
+      // supply at offset 36 (u64) -> 0
+      // decimals at offset 44 (u8)
+      data.writeUInt8(9, 44);
+      // isInitialized at offset 45 (bool)
+      data.writeUInt8(1, 45);
+      litesvm.setAccount(mint, {
+        lamports: 1_000_000, // non-zero rent to look realistic
+        data: new Uint8Array(data),
+        owner,
+        executable: false,
+        rentEpoch: 0,
+      });
+    }
+  };
+  ensure(NATIVE_MINT, TOKEN_PROGRAM_ID);
+  ensure(NATIVE_MINT_2022, TOKEN_2022_PROGRAM_ID);
+}
+
+// Normalize logs so Anchor's EventParser can decode events reliably.
+// Solana RPC emits binary event data as: "Program data: <base64>".
+// Some LiteSVM builds may emit hex or numeric arrays instead; convert them to base64.
```

Comment:
This is interesting! Could you explain more? Is behavior version dependent? Is it dependent on the underlying program?

and maybe dumb question, but is `pretty_logs` a utility that could help here? https://www.litesvm.com/docs/testing-your-program/executing-instructions#working-with-program-logs